components/webapps/blacklist-whitelist-service/src/main/java/com/wso2telco/services/bw/Queries.java

```diff
@@ -56,360 +51,262 @@
+		api = replaceAllWithWildcard(api);
+		sp = replaceAllWithWildcard(sp);
 
-				final StringBuilder succMSG = new StringBuilder();
+		blacklistWhitelistDTO.setAppId(app);
+		blacklistWhitelistDTO.setApiID(api);
+		blacklistWhitelistDTO.setAction(action);
+		blacklistWhitelistDTO.setServiceProvider(sp);
+		blacklistWhitelistDTO.setUser(user);
 
-				succMSG.append("{").append("\"Success\":").append("{").append("\"messageId\":\"")
-						.append("WhiteListed Number".intern()).append("\",");
-				succMSG.append("\"text\":\"").append("WhiteListed Number Successfully Added to the system ".intern())
-						.append("\",");
-				succMSG.append("\"variables\":\"").append(msisdn).append("\"").append("}}");
 
-				return Response.status(Response.Status.OK).entity(succMSG.toString()).build();
+		br = new BufferedReader(new InputStreamReader(uploadedInputStream, StandardCharsets.UTF_8));
 
-			}catch( InvalidMSISDNException e1){
-				return Response.status(Response.Status.BAD_REQUEST).entity(e1.getErrorType()).build();
+		try {
+			while (br.ready() && limit > 0) {
+				msisdnList.add(br.readLine());
+				limit--;
+			}
+			Response.Status status = Response.Status.CREATED;
+			blacklistWhitelistDTO.setMsisdnList(msisdnList);
+			response = blackListWhiteListService.blacklist(blacklistWhitelistDTO);
 
-			} catch (BusinessException e) {
-				return Response.status(Response.Status.BAD_REQUEST).entity(e.getErrorType()).build();
-			} catch (Exception e) {
-				e.printStackTrace();
+			if(response.getFailed() > 0 && response.getProcessed() == 0){
+				status = Response.Status.BAD_REQUEST;
 			}
-		}
-		else {
-			return Response.status(Response.Status.BAD_REQUEST).entity(errorMSG.toString()).build();
+
+			return Response.status(status).entity(gson.toJson(response)).build();
+		} catch (IOException | BusinessException e) {
+			LOG.error("Unable to process upload", e);
+			return Response.status(500).build();
 		}
 
-		return Response.ok().build();
 	}
 
-	@POST
-	@Path("/Whitelist")
-	@Consumes("application/json")
-	@Produces("application/json")
-	public Response bulkWhitelist(String jsonBody) throws SQLException {
-		LOG.debug("bulkWhitelist Triggerd  jsonBody :" + jsonBody);
-
+	/**
+	 * Removes from the blacklist
+	 *
+	 * { "apiName":"USSD" }
+	 */
+	@DELETE
+	@Path("/remove/{app}/{api}")
+	public Response removeFromBlacklist(@PathParam("app") String app, @PathParam("api") String api,
+										@QueryParam("msisdn") String msisdn, @QueryParam("action") String action, @QueryParam("sp") String sp) {
 		Gson gson = new GsonBuilder().serializeNulls().create();
-		WhiteListBulk whiteListReq = gson.fromJson(jsonBody, WhiteListBulk.class);
+		BlacklistWhitelistDTO dto = new BlacklistWhitelistDTO();
 
-		String appId = whiteListReq.getAppId();//lbs-payment-app
-		String apiId = whiteListReq.getApiId();//admin:LBS:v1.0
-		String userID = whiteListReq.getUserID();//admin
-		String[] msisdnList = whiteListReq.getMsisdnList();
+		app = replaceAllWithWildcard(app);
```

Comment:
Validate the incoming parameter before processing. Hope you are doing the same validations before sending the data into the server as well. Otherwise better to add the client-side validation as well. No point of sending requests to the server with invalid data.